### PR TITLE
fix: recurring dependency in maci context

### PR DESCRIPTION
**Pull request by kittybest**
solve #183.
The problem is explained in the issue.

### CODE REVIEW

File: src/contexts/Maci.tsx [hunk 1]
```diff
@@ -84,10 +84,33 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
...
+
+    if (a.numSignups !== b.numSignups) {
+      return true;
+    }
+
+    if (a.isStateAqMerged !== b.isStateAqMerged) {
+      return true;
+    }
+
+    return false;
```
Comment (0xmad) [inline src/contexts/Maci.tsx:112]: In the function no need to use if for every case.
```suggestion
    return a.address !== b.address || a.deployTime !== b.deployTime || a.duration !== b.duration || a.numSignups !== b.numSignups || a.isStateAqMerged !== b.isStateAqMerged;
```
And what's about using `isEqual#lodash` and `pick#lodash` specific fields for checking the data?

**Review by 0xmad — COMMENTED** [review]
@kittybest thanks, just one question

File: src/contexts/Maci.tsx (hunk 1 above)
Comment (kittybest) [reply src/contexts/Maci.tsx]: change to this: https://github.com/Ethereum-Mex/ethmx-maci-rpgf/blob/development/src/contexts/Maci.tsx#L99-L108

File: src/contexts/Maci.tsx [hunk 2]
```diff
@@ -268,7 +268,9 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
       isLoading,
       isEligibleToVote,
       initialVoiceCredits,
-      votingEndsAt,
+      votingEndsAt: pollData
```
Comment (ctrlc03) [inline src/contexts/Maci.tsx:271]: if we go with the above `useMemo` do we need this?

Comment (kittybest) [reply src/contexts/Maci.tsx:271]: you're right, I forgot to recover this part (same as the main branch)

**Review by ctrlc03 — APPROVED** [review]
Thanks

**Pull request merged**